### DISCUSSION TITLE
sudo systemctl start zigbee2mqtt  -- Won't start

### DISCUSSION THREAD

**Discussion by richiepp** [Koenkk/zigbee2mqtt#21804] [General]
Hey All,
I'm on a fresh install of the latest Raspian 64 bit and trying to get zigbee2mqtt running at reboot.  

I can get it to work using
'cd /opt/zigbee2mqtt/  
npm start'

When I try this 
`sudo systemctl start zigbee2mqtt`

I get this
`Job for zigbee2mqtt.service failed because the control process exited with error code.
See "systemctl status zigbee2mqtt.service" and "journalctl -xeu zigbee2mqtt.service" for details.`

The problem seems to be in /etc/systemd/system/zigbee2mqtt.service  Specifically this line
'ExecStart=/usr/bin/node index.js'

When I navigate to /usr/bin/ I can see the node folder but I can't navigate inside it to see index.js

I've also tried this
`ExecStart=/usr/bin/npm start`
 
How can I get this zigbee2mqtt running at reboot?
Thanks
Rich

**Comment by Nardol** [comment]
How have you installed Node?
It is strange /usr/bin/node is a folder.

At last, can you past here the content of zigbee2mqtt.service?

**Reply by richiepp** [reply]
zigbee2mqtt.service - zigbee2mqtt
     Loaded: loaded (/etc/systemd/system/zigbee2mqtt.service; enabled; preset: >
     Active: activating (auto-restart) (Result: exit-code) since Fri 2024-03-15>
    Process: 2193 ExecStart=/usr/bin/npm start (code=exited, status=1/FAILURE)
   Main PID: 2193 (code=exited, status=1/FAILURE)
        CPU: 2.424s

**Reply by Nardol** [reply]
Sorry, I was referring to your `zigbee2mqtt.service` file.

About node, can you give me the result of the following command:
```shell
command -v node
```

**Reply by richiepp** [reply]
/usr/bin/node

thx for the quick response!
R

**Reply by Nardol** [reply]
Please give me the result of the following:

```shell
cat /etc/systemd/system/zigbee2mqtt.service
```

**Comment by richiepp** [comment]
cat /etc/systemd/system/zigbee2mqtt.service
[Unit]
Description=zigbee2mqtt
After=network.target

[Service]
Environment=NODE_ENV=production
Type=notify
ExecStart=/usr/bin/npm start
WorkingDirectory=/opt/zigbee2mqtt
StandardOutput=null
# Or use StandardOutput=null if you don't want Zigbee2MQTT messages filling syslog, for more options see systemd.exec(5)
StandardError=inherit
WatchdogSec=10s
Restart=always
RestartSec=10s
User=richiep

[Install]
WantedBy=multi-user.target

**Reply by Nardol** [reply]
Please pu a `#` at the begining of the line `StandardOutput=null` then run the following commands:

```
sudo systemctl daemon-reload
sudo systemctl restart zigbee2mqtt.service
```

If it still fails, give me the result of the following command:

```
systemctl status zigbee2mqtt.service
```

**Reply by richiepp** [reply]
richiep@PermaPi:~ $ sudo systemctl daemon-reload
sudo systemctl restart zigbee2mqtt.service

Job for zigbee2mqtt.service failed because the control process exited with error code.
See "systemctl status zigbee2mqtt.service" and "journalctl -xeu zigbee2mqtt.service" for details.

richiep@PermaPi:~ $ systemctl status zigbee2mqtt.service
● zigbee2mqtt.service - zigbee2mqtt
     Loaded: loaded (/etc/systemd/system/zigbee2mqtt.service; enabled; preset: enabl>
     Active: activating (auto-restart) (Result: exit-code) since Fri 2024-03-15 14:3>
     Process: 4352 ExecStart=/usr/bin/node index.js (code=exited, status=1/FAILURE)
     Main PID: 4352 (code=exited, status=1/FAILURE)
    CPU: 2.077s
    lines 1-6/6 (END)

**Reply by Nardol** [reply]
I see some strange things:

First of all, have you put a dash character at the beginning of the line I specified in `/etc/systemd/system/zigbee2mqtt.service`?
I ask because the last command line should return more things.

Then I read the following:
``Process: 4352 ExecStart=/usr/bin/node index.js (code=exited, status=1/FAILURE)`
When you wrote the following when pasting this file content:
`ExecStart=/usr/bin/npm start`

If you had run
```
sudo systemctl daemon-reload
```

Then:
```
sudo systemctl restart zigbee2mqtt.service
```

Then finally:
```
sudo systemctl status zigbee2mqtt.service
```

I should have read something like ```Process: 4352 ExecStart=/usr/bin/npm start (code=exited, status=1/FAILURE)``.

It looks some modifications are not taken in account.

**Comment by richiepp** [comment]
Since this is a new installation, I think I'm going to reformat and try
again.

Does this sound like a good plan?
Thx!
Rich


On Fri, Mar 15, 2024 at 2:43 PM Patrick ZAJDA ***@***.***>
wrote:

> I see some strange things:
>
> First of all, have you put a dash character at the beginning of the line I
> specified in /etc/systemd/system/zigbee2mqtt.service?
> I ask because the last command line should return more things.
>
> Then I read the following:
> ``Process: 4352 ExecStart=/usr/bin/node index.js (code=exited,
> status=1/FAILURE)When you wrote the following when pasting this file
> content:ExecStart=/usr/bin/npm start`
>
> If you had run
>
> sudo systemctl daemon-reload
>
> Then:
>
> sudo systemctl restart zigbee2mqtt.service
>
> Then finally:
>
> sudo systemctl status zigbee2mqtt.service
>
> I should have read something like ```Process: 4352 ExecStart=/usr/bin/npm
> start (code=exited, status=1/FAILURE)``.
>
> It looks some modifications are not taken in account.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Koenkk/zigbee2mqtt/discussions/21804#discussioncomment-8806341>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BCKNYMMBYQDW7Y7EMD2DZG3YYM6MBAVCNFSM6AAAAABEW746U6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DQMBWGM2DC>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Ride On!
Rich Points
***@***.***
303-589-0597

**Reply by Nardol** [reply]
If you have not installed too much other things and won't loose too much, it is radical but hop it will work this time.
But before, try to see if zigbee2mqtt has created some logs in the config directory or a subdir if you can find any error.

I hop it will be better for your installation.

**Comment by richiepp** [comment]
Yes, time to reformat.
Thanks for your help.
Gratitude.
Rich

On Fri, Mar 15, 2024 at 2:50 PM Patrick ZAJDA ***@***.***>
wrote:

> If you have not installed too much other things and won't loose too much,
> it is radical but hop it will work this time.
> But before, try to see if zigbee2mqtt has created some logs in the config
> directory or a subdir if you can find any error.
>
> I hop it will be better for your installation.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Koenkk/zigbee2mqtt/discussions/21804#discussioncomment-8806408>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BCKNYMOYJQIJT23JT77GVADYYM7FXAVCNFSM6AAAAABEW746U6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DQMBWGQYDQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Ride On!
Rich Points
***@***.***
303-589-0597

**Comment by richiepp** [comment]
I'm back!  LOL
So I reformatted and tried again and I'm getting the exact same error
message.

sudo systemctl start zigbee2mqtt

Job for zigbee2mqtt.service failed because the control process exited with
error code.
See "systemctl status zigbee2mqtt.service" and "journalctl -xeu
zigbee2mqtt.service" for details.

###
systemctl status zigbee2mqtt.service
    Process: 3956 ExecStart=/usr/bin/node index.js (code=exited,
status=1/FAILURE)
###
  journalctl -xeu zigbee2mqtt.service
 An ExecStart= process belonging to unit zigbee2mqtt.service has exited.
░░
░░ The process' exit code is 'exited' and its exit status is 1.
Mar 15 17:44:37 PermaPi systemd[1]: zigbee2mqtt.service: Failed with result
'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ The unit zigbee2mqtt.service has entered the 'failed' state with result
'exit-code'.
Mar 15 17:44:37 PermaPi systemd[1]: Failed to start zigbee2mqtt.service -
zigbee2mqtt.
░░ Subject: A start job for unit zigbee2mqtt.service has failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ A start job for unit zigbee2mqtt.service has finished with a failure.
░░
░░ The job identifier is 6108 and the job result is failed.
Mar 15 17:44:37 PermaPi systemd[1]: zigbee2mqtt.service: Consumed 1.986s
CPU time.
░░ Subject: Resources consumed by unit runtime
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ The unit zigbee2mqtt.service completed and consumed the indicated
resources.

###
more details

   - Raspberry Pi 5
   - 4GB memory
   - using the raspberry pi image loader to format
   - 64gb sd card
   - 5A power supply

I'm following the zigbee2mqtt installation directions as closely as I can.


On Fri, Mar 15, 2024 at 3:02 PM richpoints ***@***.***> wrote:

> Yes, time to reformat.
> Thanks for your help.
> Gratitude.
> Rich
>
> On Fri, Mar 15, 2024 at 2:50 PM Patrick ZAJDA ***@***.***>
> wrote:
>
>> If you have not installed too much other things and won't loose too much,
>> it is radical but hop it will work this time.
>> But before, try to see if zigbee2mqtt has created some logs in the config
>> directory or a subdir if you can find any error.
>>
>> I hop it will be better for your installation.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Koenkk/zigbee2mqtt/discussions/21804#discussioncomment-8806408>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/BCKNYMOYJQIJT23JT77GVADYYM7FXAVCNFSM6AAAAABEW746U6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DQMBWGQYDQ>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>
>
> --
> Ride On!
> Rich Points
> ***@***.***
> 303-589-0597
>


-- 
Ride On!
Rich Points
***@***.***
303-589-0597

**Reply by Nardol** [reply]
> I'm following the zigbee2mqtt installation directions as closely as I can.

If there has been some differences, can you tell what are these?

Have you checked Zigbee2MQTT logs?

Have you also tried again to comment (putting a dash character at the beginning) the line I suggested you?
It could be really helpful to have more verbosity from the SystemD service.

Each time you modify the file `/etc/systemd/system/zigbee2mqtt.service`, don't forget to run
```shell
sudo systemctl daemon-reload
```
Before restarting the service else it won't be updated.

**Comment by lux73** [comment]
does it work if you execute `npm start` inside your zigbee2mqtt folder?

if not there is no need to fight with zigbee2mqtt.service 😉

**Comment by richiepp** [comment]
yes, it works with npm start

**Reply by lux73** [reply]
show `ls -al` inside the zigbee2mqtt Folder

**Comment by richiepp** [comment]
SOLVED  

Turns out I don't know how to chown

in the zigbee2mqtt installation guide it says to
`sudo chown -R ${USER}: /opt/zigbee2mqtt`

so I wrote the following which didn't work and explains why I had to use sudo for the rest of the installation
`sudo chown -R ${pi}: /opt/zigbee2mqtt`

I used this and it worked!
'sudo chown -R pi:pi /opt/zigbee2mqtt'

**Reply by lux73** [reply]
so it was a permission Problem ^^

**Reply by Nardol** [reply]
A permission issue which could have been seen within changing the line I suggested because we would have read there was permission issue :slightly_smiling_face: 

In this case, `sudo chown -R ${USER}: /opt/zigbee2mqtt` meant literally `sudo chown -R ${USER}: /opt/zigbee2mqtt`, nothing to replace, `${USER}` is an environment variable which is set to your user name.
I understand the confusion in the instructions if you are not familiar with Linux.
